### PR TITLE
add "saveIfNotExists" behavior to UserExperimentRepo

### PR DESCRIPTION
**Pull request by camhashemi**
no changes to logic, just adding this for future use

@ryanpbrewster @andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserExperimentRepo.scala [hunk 1]
```diff
@@ -94,4 +95,10 @@ class UserExperimentRepoImpl @Inject() (
         (UserExperimentType(name), count)
     }
   }
+
+  def ensureUserHasExperiment(userId: Id[User], experimentType: UserExperimentType)(implicit session: RWSession): Unit = {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserExperimentRepo.scala:99]: Repos should return something.

**Comment by andrewconner** [comment]
Check out `UserExperimentCommander.addExperimentForUser`. Does what you want, and doesn't break dynamic experiments.

**Comment by camhashemi** [comment]
I don't see that method there. Do you mean `addDynamicExperiments`?

**Comment by andrewconner** [comment]
Sorry, it's in shoebox's implementation. `LocalUserExperimentCommander`

**Comment by camhashemi** [comment]
nice, wish it was used more! thank you.

**Pull request closed without merging**